binary_gentoo/internal/cli/build.py

```diff
@@ -192,10 +198,19 @@ def build(config):
     ]
 
     # Create pretend log dir
-    category, package = extract_category_package_from(config.atom)
+    try:
+        emerge_target_type = EmergeTargetType.PACKAGE
+        category, package_or_set = extract_category_package_from(config.emerge_target)
+    except ValueError as package_error:
+        try:
+            emerge_target_type = EmergeTargetType.SET
+            category, package_or_set = 'sets', extract_set_from(config.emerge_target)
+        except ValueError as set_error:
```

Comment:
How about extracting a method `emerge_target_type, category, package_or_set = classify_emerge_target(config.emerge_target)` at this very place? The function does way too much even before this addition, and I think that extraction would help readability.

Reply:
yes that is indeed a good idea